batch/batch/driver/instance_pool.py

```diff
@@ -129,6 +70,54 @@ def __init__(self, app, machine_name_prefix):
 
         self.name_instance = {}
 
+    async def update_zones(self):
+        northamerica_regions = {
+            # 'northamerica-northeast1',
+            'us-central1',
+            'us-east1',
+            'us-east4',
+            'us-west1',
+            'us-west2',
+            'us-west3',
+            'us-west4'
+        }
+
+        new_zones = []
+        new_zone_weights = []
+
+        async for r in await self.compute_client.list('/regions'):
+            name = r['name']
+            if name not in northamerica_regions:
+                continue
+
+            quota_remaining = {
+                q['metric']: q['limit'] - q['usage']
```

Comment:
oh cool. That's nice they give you both the quota and the usage.

Reply:
Yeah.  I started with another API, the Service Usage API (https://cloud.google.com/service-usage/docs/manage-quota), but it only gives you limits.  Finding this was a major simplification.